packages/material-ui/src/styles/useThemeProps.d.ts

```diff
@@ -8,8 +10,16 @@ export type ThemedProps<Theme, Name extends keyof any> = Theme extends {
   ? Props
   : {};
 
+export interface AdditionalThemeProps<Theme> {
+  isRtl: boolean;
+  theme: Theme;
```

Comment:
Added a test that asserts the exact types. Especially for the `Theme` it is quite important that we get the `Theme` we declared in the type parameter not just any theme extending `ThemeWithProps`.

Reply:
Wasn't aware of the `expectType`, thanks for updating the test

Reply:
I expected that the Theme would be the same as useTheme or provided by generic type.

```
import { Theme } from './createMuiTheme';
import { Components } from './components';

export interface ThemeWithProps {
  components?: Components;
}

export type ThemedProps<T, Name extends keyof any> = T extends {
  components: Record<Name, { defaultProps: infer Props }>;
}
  ? Props
  : {};

export interface AdditionalThemeProps<T> {
  isRtl: boolean;
  theme: T;
}

export default function useThemeProps<
  T extends ThemeWithProps = Theme,
  Props = {},
  Name extends keyof any = any
>(params: {
  props: Props;
  name: Name;
}): Props & ThemedProps<T, Name> & AdditionalThemeProps<T>;
```

Reply:
Do you mean you expected that the generic will have default value? I don’t have strong opinion on this, but wouldn’t be against it...

Reply:
Yes, because useThemeProps uses material-ui/src/styles/useTheme.d.ts
```
import { Theme } from './createMuiTheme';

export default function useTheme<T = Theme>(): T;
```
 So the response should be the same theme I think.

Reply:
@povilass great, would you be willing to open a PR?

Reply:
@mnajdova  Ok gonna make it later.